test/test_setmacro.jl

```diff
@@ -45,5 +45,9 @@ using StaticArrays: @SMatrix
     @test m3 === @SMatrix[1 0; 0 0]
 end
 
+@testset "issue #83" begin
+    @test_throws LoadError eval(:(Setfield.@lens(_.[:a])))
```

Comment:
That didn't work, because the error happens during macro expansion

Reply:
I think using `@eval` for testing macro expansion is an OK approach.  But we can use `lensmacro` function in this case.  It would be better since the exception won't be wrapped by `LoadError`.

Reply:
Ah right, sorry for the noise!